src/modules/commons/components/searchableSelect/SearchableSelect.js

```diff
@@ -0,0 +1,303 @@
+/**
+ * @module modules/commons/components/searchableSelect/SearchableSelect
+ */
+
+import css from './searchableSelect.css';
+import { html } from 'lit-html';
+import { MvuElement } from '../../../MvuElement';
+import { isNumber } from '../../../../utils/checks';
+import { KeyActionMapper } from '../../../../utils/KeyActionMapper';
+
+const Update_Placeholder = 'update_placeholder';
+const Update_Options = 'update_options';
+const Update_Selected = 'update_selected';
+const Update_Search = 'update_search';
+const Update_MaxEntries = 'update_maxEntries';
+
+/**
+ * General purpose implementation of a select-like component with integrated filtering.
+ *
+ * @property {string} placeholder='' - The placeholder to show when the search field is empty.
+ * @property {number} maxEntries=10 - The maximum amount of entries to show in the select field.
+ * @property {string|null} selected=null - The currently selected option.
+ * @property {string} search='' - The search term to filter through the provided options
+ * @property {Array<String>} options - Unfiltered options the user can choose from
+ * @property {function(changedState)} onChange - The Change callback function when the search input changes or an option is selected
+ * @fires onChange Fires when the search input changes or an option is selected
+ *
+ * @class
+ * @author herrmutig
+ */
+export class SearchableSelect extends MvuElement {
+	#hasPointer;
+	#keyActionMapper;
+
+	#onPointerCancelActionListener = () => {
+		if (this.#hasPointer) return;
+		this.#cancelAction();
+	};
+
+	// eslint-disable-next-line no-unused-vars
+	#onChange = (changedState) => {};
+
+	constructor() {
+		super({
+			placeholder: 'Search...',
+			maxEntries: 10,
+			selected: null,
+			search: '',
+			options: [],
+			availableOptions: []
+		});
+
+		this.#hasPointer = false;
+	}
+
+	onInitialize() {
+		document.addEventListener('click', this.#onPointerCancelActionListener);
+
+		this.#keyActionMapper = new KeyActionMapper(document);
+		this.#keyActionMapper.deactivate();
+
+		this.observeModel(['selected', 'search'], () => this.#dispatchChangeEvents());
```

Comment:
Let's try to separate the different changes
1. The search term changed -> we fire a `search` CustomEvent 
2. An entry was selected -> we fire a `select` CustomEvent